docs/source/developer_documentation.md

```diff
@@ -13,7 +13,7 @@
 
 ## Installation for Developers
 
-- `poetry install --with dev`
+- `uv pip install -r pyproject.toml --extra dev`
```

Comment:
I notice this command used here and there in this PR. I think it could be replaced by `uv sync --all-groups` which creates the `uv.lock` and syncs the environment with all primary dependencies and all dependency groups.